en/identity-server/7.0.0/docs/guides/authentication/conditional-auth/typingdna-based-adaptive-auth.md

```diff
@@ -0,0 +1,121 @@
+
+7. Click **Update** to save the changes.
+
+## Configure TypingDNA in applications
+
+Follow the steps given below to configure TypingDNA in your application.
+
+1. Go to **Service Providers -> List** and click **Edit** on the service provider that you want to configure TypingDNA in.
+
+2. Expand **Local and Outbound Authentication Configuration** and click **Advanced Configuration**.
+3. Configure two authentication steps.
+
+    !!! info
+        In this scenario, we will configure **Username and Password** and **TOTP**.
+    
+    ![TypingDNA configure two authentication steps]({{base_path}}/assets/img/samples/typingdna-two-steps.png)
+
+4. Expand **Script Based Adaptive Authentication** and add the following script:
+
+    ``` js
+    // This script will step up 2FA authentication if the user's typing behaviour does not match with the enrolled behaviour.
+
+    // You can use the parameters 'score' (num 0-100), 'result' (boolean), 'confidence' (num 0-100), 'comparedPatterns' in your 
+    // authentication logic to trigger the 2nd step. 
+    // Only the 'result' parameter has been used in the sample script. 
+
+    var onLoginRequest = function(context) {
+        executeStep(1, {
+            onSuccess: function (context) {
+                verifyUserWithTypingDNA(context, {
+                    onSuccess: function(context,data){
+                        // Change the definition here as required.
+                        var userVerified = data.result;
+
+                        // data.isTypingPatternReceived indicates whether a typing pattern is received from the login portal.
+                        if (data.isTypingPatternReceived && !userVerified){
+                            executeStep(2);
+                        }
+                    },onFail: function(context,data){
+                        executeStep(2);
+                    }
+                });
+            }
+        });
+    };
+
+    ```
+    ![TypingDNA script]({{base_path}}/assets/img/samples/typingdna-script.png)
+
+
+5. Click **Update** to save the changes.
+
+## Try it out
+
+1. Access the login page of the sample application and click **Log in**
+2. Use the credentials of **Alex** and log in to the application two times.
+    
+    !!! info
+        You will be prompted for the second step on both occasions. TypingDNA requires two initial enrollments to register the user’s typing pattern. You can change the number of minimum initial enrollments required in the API settings of the [typingDNA dashboard](https://www.typingdna.com/clients/).
+
+3. Log in for the third time with Alex's credentials.
+
+    !!! info
+        From this log in attempt and beyond, typingDNA will analyze your typing pattern against the registered typing pattern of the account. TOTP will only be prompted if your typing pattern does not match the typing pattern registered in Alex's account.
+
+
+
+
+
```

Comment:
shall we remove additional new lines?

Reply:
Fixed with https://github.com/wso2/docs-is/pull/4732/commits/bb12fe62bc6ccf92b2116ae0182ef4ff3ae6a9c7